### ISSUE TITLE
Typescript 3.7.x breaking @aws-sdk/types

### ISSUE THREAD

**Issue by alexforsyth** [aws/aws-sdk-js-v3#1919]
**Describe the bug**
`@aws-sdk/types` declares UserAgentPair.
```ts
export declare type UserAgentPair = [name: string, version?: string];
```

Pairs are not supported in Typescript 3.7.x
https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#labeled-tuple-elements.

Users running typescript < 4.0.0 cannot correctly get types. 

**SDK version number**
`3.3.0`

**Is the issue in the browser/Node.js/ReactNative?**
All

**Details of the browser/Node.js/ReactNative version**
node v14.15.4

**To Reproduce (observed behavior)**

The following error is thrown for the code example give below:
```console
$ yarn build
yarn run v1.22.5
$ tsc
node_modules/@aws-sdk/types/dist/cjs/util.d.ts:92:42 - error TS1005: ',' expected.

92 export declare type UserAgentPair = [name: string, version?: string];
                                            ~

node_modules/@aws-sdk/types/dist/cjs/util.d.ts:92:60 - error TS1005: ',' expected.

92 export declare type UserAgentPair = [name: string, version?: string];
                                                              ~


Found 2 errors.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

<details>
<summary>package.json</summary>

```json
{
  "name": "test-ts-3.7",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "build": "tsc"
  },
  "dependencies": {
    "@aws-sdk/client-s3": "3.3.0"
  },
  "devDependencies": {
    "@types/node": "14.14.22",
    "typescript": "3.7.5"
  }
}
```

</details>

<details>
<summary>index.ts</summary>

```ts
import { S3 } from "@aws-sdk/client-s3";

(async () => {
  const region = "us-west-2";
  const client = new S3({ region });
  await client.listBuckets({});
})();

```

</details>

<details>
<summary>tsconfig.json</summary>

```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "strict": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true
  }
}
```

</details>

**Expected behavior**
No error is thrown when compiling client code with TypeScript version 3.7.x

**Comment by trivikr** [comment]
This issue would be fixed by using [downlevel-dts](https://www.npmjs.com/package/downlevel-dts) to create type definition files which support TS 3.4 or later.

**Comment by trivikr** [comment]
Three PRs which need to be merged before TS 3.4 compatible types can be generated:
* https://github.com/aws/aws-sdk-js-v3/pull/1936
* https://github.com/aws/aws-sdk-js-v3/pull/1937
* https://github.com/aws/aws-sdk-js-v3/pull/1938

Experimental commits are in downlevel-dts branch https://github.com/trivikr/aws-sdk-js-v3/tree/downlevel-dts

**Comment by ohana54** [comment]
@trivikr I upgraded to 3.4.0 and it still doesn't work - I use TS 3.9.7, while labeled tuples were added in 4.x. It's probably due to this condition:
https://github.com/aws/aws-sdk-js-v3/blob/63ad2151c8bb7be32ea8838a9b0974806ed3906b/packages/types/package.json#L29

If I change it to `<=` it works fine.

**Comment by trivikr** [comment]
> I upgraded to 3.4.0 and it still doesn't work - I use TS 3.9.7, while labeled tuples were added in 4.x. It's probably due to this condition:

Can you create a separate issue specific to 3.9.7?
I'll provide a fix, and also update downlevel-dts README https://github.com/sandersn/downlevel-dts#usage

**Comment by ohana54** [comment]
Thanks! I can open a PR for the readme
